File: adventure/src/main/java/x/withlithum/neoware/adventure/content/AdventureContentTreeFactory.java
/*
 * SPDX-FileCopyrightText: 2025-2026 WithLithum & contributors
 * SPDX-License-Identifier: AGPL-3.0-or-later
 */

package x.withlithum.neoware.adventure.content;

import x.withlithum.neoware.data.content.hierarchy.ContentTreeFactory;
import x.withlithum.neoware.util.results.NeoResult;

import java.nio.file.Path;

public final class AdventureContentTreeFactory implements ContentTreeFactory<AdventureContentTree> {
    private AdventureContentTreeFactory() {
    }

    public static final AdventureContentTreeFactory INSTANCE = new AdventureContentTreeFactory();

    @Override
    public AdventureContentTree getEmpty() {
        return AdventureContentTree.EMPTY;
    }

    @Override
    public int getDataVersion() {
        return 1;
    }

    @Override
    public NeoResult<AdventureContentTree> loadDir(Path directory) {
        return AdventureContentTree.loadDir(directory);
    }
}

File: adventure/src/main/kotlin/x/withlithum/neoware/adventure/content/AdventureContentTreeFactory.kt
/*
 * SPDX-FileCopyrightText: 2025-2026 WithLithum & contributors
 * SPDX-License-Identifier: AGPL-3.0-or-later
 */

package x.withlithum.neoware.adventure.content

import okio.FileSystem
import okio.Path.Companion.toOkioPath
import x.withlithum.neoware.data.content.hierarchy.ContentTreeFactory
import x.withlithum.neoware.util.results.NeoResult
import java.nio.file.Path

object AdventureContentTreeFactory : ContentTreeFactory<AdventureContentTree> {
    override fun getEmpty() = AdventureContentTree.EMPTY
    override fun getDataVersion() = 1

    override fun loadDir(
        dir: Path
    ): NeoResult<AdventureContentTree> {
        return AdventureContentTree.loadDir(dir.toOkioPath(), FileSystem.SYSTEM)
    }
}